src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditStatementTests.cs

```diff
@@ -7089,5 +7089,143 @@ class C
         }
 
         #endregion
+
+        #region C# 7
+
+        [Fact]
+        public void DeconstructionStatement()
+        {
+            string src1 = @"
+var (x1, x2) = (1, 2);
+";
+
+            string src2 = @"
+var (x3, x4) = (1, 2);
+var (x1, x2) = (30, 40);
+";
+
+            // variable names weight more in matching
+            var expected = new MatchingPairs
+            {
+                { "var (x1, x2) = (1, 2);", "var (x1, x2) = (30, 40);" },
+                { "var (x1, x2) = (1, 2)", "var (x1, x2) = (30, 40)" }
+            };
+
+            var match = GetMethodMatch(src1, src2);
+            var actual = ToMatchingPairs(match);
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void ForEachComponentStatement()
+        {
+            string src1 = @"
+foreach(var (x1, x2) in new[] { (1, 2) }) { Console.Write(1); }
+";
+
+            string src2 = @"
+foreach(var (x3, x4) in new[] { (1, 2) }) { Console.Write(1); }
+foreach(var (x1, x2) in new[] { (30, 40) }) { Console.Write(50); }
+";
+
+            // variable names weight more in matching
+            var expected = new MatchingPairs
+            {
+                { "foreach(var (x1, x2) in new[] { (1, 2) }) { Console.Write(1); }", "foreach(var (x1, x2) in new[] { (30, 40) }) { Console.Write(50); }" },
+                { "{ Console.Write(1); }", "{ Console.Write(50); }" },
+                { "Console.Write(1);", "Console.Write(1);" }
+            };
```

Comment:
The same thing happens with regular `foreach` statement (I'll add a test for that).
The reason is that `StatementSyntaxComparer.Classify` labels `ExpressionStatement` rather than ignores it.